### PR TITLE
Fix jlink multiple frame read

### PR DESCRIPTION
**Pull request by anwu**
### event_loop.py
- Allow `run_executor` to be run in coroutine finalizer function
### jlink.py
- Changed `disconnect` logic to close "rpc" interface last
- Added task to handle when iotile is `quit` to cleanup nicely
### jlink_background.py
- Changed frame reading logic from one at a time to all frames available in queue at once to minimize jlink memory read/writes
- Performance enhancement by removing delays
- Fixed coroutine quitting issues because of while loop in `maintenance_coroutine`

### CODE REVIEW

File: iotilecore/iotile/core/utilities/async_tools/event_loop.py [hunk 1]
```diff
@@ -288,10 +288,9 @@ def start(self, aug='EventLoopThread'):
         running, it will not do anything.
         """
 
-        if self.stopping:
-            raise LoopStoppingError("Cannot perform action while loop is stopping.")
-
         if not self.loop:
+            if self.stopping:
+                raise LoopStoppingError("Cannot perform action while loop is stopping.")
             self._logger.debug("Starting event loop")
```
Comment (timburke) [inline iotilecore/iotile/core/utilities/async_tools/event_loop.py]: Missing blank line

File: transport_plugins/jlink/iotile_transport_jlink/jlink_background.py [hunk 2]
```diff
@@ -322,21 +322,23 @@ def _connect_jlink_blocking(jlink_serial, jlink_name):
         return self._jlink
 
     async def close_jlink(self):
+        if self._maintenance_task is not None:
+            await self._maintenance_task.stop()
         if self._maintenance_counter:
```
Comment (timburke) [inline transport_plugins/jlink/iotile_transport_jlink/jlink_background.py]: Missing blank line

File: iotilecore/iotile/core/utilities/async_tools/event_loop.py (hunk 1 above)
Comment (anwu) [reply iotilecore/iotile/core/utilities/async_tools/event_loop.py]: Fixed

File: transport_plugins/jlink/iotile_transport_jlink/jlink_background.py (hunk 2 above)
Comment (anwu) [reply transport_plugins/jlink/iotile_transport_jlink/jlink_background.py]: Fixed

**Pull request merged**